ci/scripts/update-mainnet-artifacts.sh

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
+
+# This script is run by the update-mainnet-artifacts job in the ci-pr-only workflow
+# to update ./mainnet-artifacts.bzl bazed on the contents of testnet/mainnet_revisions.json.
+# The latter tracks the IC version (git revision) of the mainnet NNS subnet (with principal tdb26-...).
+#
+# This script will set the MAINNET_NNS_SUBNET_IC_VERSION bazel variable and will compute the SHA256 hashes
+# of the PUBLISHED_BINARIES which are using the MAINNET_NNS_SUBNET_IC_VERSION.
+
+set -eufo pipefail
+
+EXIT_STATUS=0
+
+# Retrieve the IC version (git revision) of the mainnet NNS subnet:
+MAINNET_NNS_SUBNET_ID="tdb26-jop6k-aogll-7ltgs-eruif-6kk7m-qpktf-gdiqx-mxtrf-vb5e6-eqe"
```

Comment:
Is this a placeholder? Or why is it set this way?

Reply:
It's not a placeholder. It's the ID of the NNS subnet on mainnet. See: https://github.com/dfinity/ic/blob/master/testnet/mainnet_revisions.json#L3.

Reply:
Also: https://dashboard.internetcomputer.org/subnet/tdb26-jop6k-aogll-7ltgs-eruif-6kk7m-qpktf-gdiqx-mxtrf-vb5e6-eqe

Reply:
Hmm... maybe I'm missing something obvious, but how does it get updated then? And why doesn't it just directly pull from the file you referenced instead of hardcoding it?

Reply:
The way it works is as follows:

* In the release repo there's the job [update-ic-versions-file](https://github.com/dfinity-ops/release/blob/main/.github/workflows/update-ic-versions-file.yaml) which makes a PR in the ic repo updating `testnet/mainnet_revisions.json` every time the NNS subnet on mainnet is upgraded.

* What this PR is doing is adding a CI job that updates `mainnet-artifacts.bzl` when  `testnet/mainnet_revisions.json` is modified. 

* It's unfortunately not possible to directly import `testnet/mainnet_revisions.json` into Bazel (I miss Nix so much ;) ) which is why we have to do this via this convoluted way.

Reply:
Okay, that all makes sense, but then which process updates `update-mainnet-artifacts.sh` when the version changes?

Reply:
The subnet ID never changes. `tdb26-...` has been the NNS subnet since genesis.

Reply:
Or maybe I misunderstood the question. WDYM with "version"? Version of what?

Reply:
Ah okay never mind, I was confused - this makes sense now 👍